simplex/block.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package simplex
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"sync"
+
+	"github.com/ava-labs/simplex"
+
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+)
+
+type VerifiedBlock struct {
+	computeDigestOnce sync.Once
+	digest            simplex.Digest // cached, not serialized
+
+	metadata   simplex.ProtocolMetadata
+	innerBlock []byte
+}
+
+// BlockHeader returns the block header for the verified block.
+func (v *VerifiedBlock) BlockHeader() simplex.BlockHeader {
+	v.computeDigestOnce.Do(v.computeDigest)
+	return simplex.BlockHeader{
+		ProtocolMetadata: v.metadata,
+		Digest:           v.digest,
+	}
+}
+
+// Bytes returns the serialized bytes of the verified block.
+// It concatenates the metadata bytes followed by the inner block bytes.
+func (v *VerifiedBlock) Bytes() []byte {
+	mdBytes := v.metadata.Bytes()
+	buff := make([]byte, len(mdBytes)+len(v.innerBlock))
+	copy(buff, mdBytes)
+	copy(buff[len(mdBytes):], v.innerBlock)
+	return buff
+}
+
+// computeDigest computes the digest of the block.
+func (v *VerifiedBlock) computeDigest() {
+	mdBytes := v.metadata.Bytes()
+	h := sha256.New()
+	h.Write(mdBytes)
+	h.Write(v.innerBlock)
+	digest := h.Sum(nil)
+	v.digest = simplex.Digest(digest)
+}
+
+type blockDeserializer struct {
+	vm block.ChainVM
```

Comment:
I don't think we need to drag the entire ChainVM interface here just to parse the block. Can't we just define the interface we need?